src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -254,29 +351,44 @@ static void CheckForExperimentStatus(ITextView textView, in CompletionOptions op
             {
                 AsyncCompletionLogger.LogExpanderUsage();
 
+                // It's possible we didn't provide expanded items at the beginning of completion session because it was slow even if the feature is enabled.
+                // ExpandedItemsTask would be available in this case, so we just need to return its result.
+                // We are doing this in a spinning fashion because the lifetime of expandedItemsTask is associated with the completion session, we don't
+                // want to cancel that task when this "GetExpandedCompletionContextAsync" operation is being canceled, so we will still be able to retrieve
+                // the results later.
+                if (session.Properties.TryGetProperty<Task<(AsyncCompletionData.CompletionContext, CompletionList)>>(ExpandedItemsTask, out var expandedItemsTask))
+                {
+                    // Make sure the task is removed when returning expanded items,
+                    // so duplicated items won't be added in subsequent list updates.
+                    session.Properties.RemoveProperty(ExpandedItemsTask);
+                    var (expandedContext, expandedCompletionList) = await expandedItemsTask.ConfigureAwait(false);
+                    AddPropertiesToSession(session, expandedCompletionList, triggerLocation: null);
+                    return expandedContext;
+                }
+
+                // We only reach here when expanded items are disabled, but user requested them explicitly via expander.
+                // In this case, enable expanded items and trigger the completion only for them.
```

Comment:
would it be simpler to *always* kick off the expanded items work in the beginning?  i get that it means we're potentially always doing more work.  but is that such a big deal if it's just optionally happening in the BG.  Most users will still have that work be done, esp. if we expect the defualt to be that this is always on.  So while it's an optimization opportunity, it's not necessarily that important given that it would still be running most of hte time for most people.

Reply:
Yes, it will make it simpler, but not really that much (~10 fewer lines of code). Given how rare expander button is used based on telemetry, I think it's better to avoid that work for people who don't have import completion enabled, even if it's a small group.

Reply:
i don't think it's a questino about the expander being rarely used.  presumably we want to get to the point that the *default* is that we include symbols from unimported namespaces.  so the majority of people will have that behavior.

so this complexity is only about making perf a little better for the vast minority that won't have this on.